mongo-oplog/index.d.ts

```diff
@@ -0,0 +1,19 @@
+// TypeScript Version: 2.1
+// Type definitions for mongo-oplog v2.0.2
+// Project: https://github.com/cayasso/mongo-oplog
+// Definitions by: FinalDes <https://github.com/FinalDes>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="eventemitter3" />
+
+export function MongoOplog(url: string, filter: any): OplogTrigger;
+
+export interface OplogTrigger {
+    tail(): any; // return promise
+    on(type: string, callback: (doc: any) => void): any; // return eventemitter3
+    stop(callback: () => void): any; // return promise
+}
+
+declare module "mongo-oplog" {
+    export = MongoOplog;
```

Comment:
If the module export value is a function, use [this](http://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-function-d-ts.html) pattern: `declare function MongoOplog`, then have `declare namespace MongoOplog { interface OplogTrigger { ... } }` and finally `export = MongoOplog` (with no `declare module` wrapper).